framework/src/meshgenerators/ExtraNodesetGenerator.C

```diff
@@ -74,15 +62,16 @@ ExtraNodesetGenerator::generate()
   BoundaryInfo & boundary_info = mesh->get_boundary_info();
 
   // add nodes with their ids
-  for (const auto & node_id : nodes)
-  {
-    // Our mesh may be distributed and this node may not exist on this process
-    if (!mesh->query_node_ptr(node_id))
-      continue;
+  if (isParamValid("nodes"))
```

Comment:
this code would behave fine with a nodes {} default and you would not have such a large diff either

Reply:
No, if you read the changing part, using 'IsValidParam' here is much better than has a bool variable flag and check.
```
 // make sure the input is not empty
  bool data_valid = false;
  if (isParamValid("nodes") && nodes.size() != 0)
    data_valid = true;
  if (_pars.isParamValid("coord") && coord.size() != 0)
    data_valid = true;
 ```

Reply:
I don't get this either. Why is isParamValid necessary here? If nodes is empty the for loop is skipped automatically. I'm with Guillome. #istandwithguillaume

Reply:
In the previous code, it has a mooseError to remind people the "node can not be empty". `isParamValid` will give similar error. Whoever wrote it want to have the error, I just preserve that logic instead of being silence

Reply:
you deleted those checks and there's no size checking now

Reply:
`isParamValid` gives the same thing, I don't use default value here, now if user don't provide default value, it will error out